src/utils/fromJS.js

```diff
@@ -8,6 +8,31 @@
 
 const Immutable = require("immutable");
 
+/*
+  creates an immutable map, where each of the value's
+  items are transformed into their own map.
+
+  NOTE: we guard agains `length` being a property because
+  length confuses Immutable's internal algorithm.
+*/
+function createMap(value) {
+  if (value.hasOwnProperty("length")) {
+    value.length = `${ value.length}`;
```

Comment:
I don't think createMap should have a side-effect on the value argument passed to it.

We should either restore `value.length = parseInt(value.length, 10);`, 
or work on a copy of value `copy = Object.assign({}, value, { length: undefined });`